pkg/provider/cloud/openstack/helper.go

```diff
@@ -56,6 +56,49 @@ const (
 	resourceNamePrefix = "kubernetes-"
 )
 
+func getRouterByName(netClient *gophercloud.ServiceClient, name string) (*osrouters.Router, error) {
+	routers, err := getAllRouters(netClient, osrouters.ListOpts{Name: name})
+	if err != nil {
+		return nil, err
+	}
+	switch len(routers) {
+	case 1:
+		return &routers[0], nil
+	case 0:
+		return nil, fmt.Errorf("router with name '%s' not found", name)
+	default:
+		return nil, fmt.Errorf("found %d router for name '%s', expected exactly one", len(routers), name)
+	}
+}
+
+func getRouterByID(netClient *gophercloud.ServiceClient, id string) (*osrouters.Router, error) {
+	routers, err := getAllRouters(netClient, osrouters.ListOpts{ID: id})
+	if err != nil {
+		return nil, err
+	}
+	switch len(routers) {
+	case 1:
+		return &routers[0], nil
+	case 0:
+		return nil, fmt.Errorf("router with ID '%s' not found", id)
+	default:
+		return nil, fmt.Errorf("found %d router for ID '%s', expected exactly one", len(routers), id)
+	}
+}
+
+func getAllRouters(netClient *gophercloud.ServiceClient, listOpts osrouters.ListOpts) ([]osrouters.Router, error) {
```

Comment:
`listRouters` or so might be a better name, the "_all_" confusese me, because in both cases where this function is used, were are not listing _all_.

Reply:
done